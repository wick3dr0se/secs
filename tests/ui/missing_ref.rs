use secs::World;

fn optional_components() {
    let world = World::default();

    let mut results = vec![];
    world.query(|_, _: &u32, _: &str| {});
    //~^ ERROR: is not a valid query

    world.query(|_, _: u32, _: &&str| {});
    //~^ ERROR: is not a valid query
}
