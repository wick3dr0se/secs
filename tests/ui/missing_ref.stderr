error[E0277]: `{closure@tests/ui/missing_ref.rs:7:17: 7:38}` is not a valid query
   --> tests/ui/missing_ref.rs:7:17
    |
7   |     world.query(|_, _: &u32, _: &str| {});
    |           ----- ^^^^^^^^^^^^^^^^^^^^^^^^
    |           |
    |           required by a bound introduced by this call
    |
    = help: the trait `secs::query::Query<_>` is not implemented for closure `{closure@tests/ui/missing_ref.rs:7:17: 7:38}`
    = note: only tuples with 1 or up to 5 elements can be used as queries
note: required by a bound in `World::query`
   --> $DIR/src/world.rs
    |
LLL |     pub fn query<Q: Query<T>, T>(&self, f: Q) {
    |                     ^^^^^^^^ required by this bound in `World::query`

error[E0277]: `{closure@tests/ui/missing_ref.rs:10:17: 10:38}` is not a valid query
   --> tests/ui/missing_ref.rs:10:17
    |
10  |     world.query(|_, _: u32, _: &&str| {});
    |           ----- ^^^^^^^^^^^^^^^^^^^^^^^^
    |           |
    |           required by a bound introduced by this call
    |
    = help: the trait `secs::query::Query<_>` is not implemented for closure `{closure@tests/ui/missing_ref.rs:10:17: 10:38}`
    = note: only tuples with 1 or up to 5 elements can be used as queries
note: required by a bound in `World::query`
   --> $DIR/src/world.rs
    |
LLL |     pub fn query<Q: Query<T>, T>(&self, f: Q) {
    |                     ^^^^^^^^ required by this bound in `World::query`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0277`.
